workspaces/global-floating-action-button/plugins/global-floating-action-button/src/components/FABWithSubmenu.tsx

```diff
@@ -43,36 +55,57 @@ export const FABWithSubmenu = ({
   }, [pathname]);
 
   const handleClick = () => {
-    setIsMenuOpen(!isMenuOpen);
+    setIsMenuOpen(prev => !prev);
   };
   return (
     <>
-      <Tooltip title="Menu" placement="left">
+      <Tooltip
+        title="Menu"
+        placement={slot === Slot.PAGE_END ? 'left' : 'right'}
+      >
         <Fab
           size="medium"
-          color="default"
+          color="info"
           onClick={handleClick}
           aria-label="Menu"
           variant="circular"
+          data-testid="fab-with-submenu"
         >
-          {isMenuOpen ? <CloseIcon /> : <MenuIcon />}
+          {isMenuOpen ? (
+            <CloseIcon className={styles.menuButtonStyle} />
+          ) : (
+            <MenuIcon className={styles.menuButtonStyle} />
+          )}
         </Fab>
       </Tooltip>
-      {isMenuOpen && (
-        <Slide
-          container={ref}
-          easing={{
-            enter: theme.transitions.easing.easeOut,
-            exit: theme.transitions.easing.sharp,
-          }}
-        >
-          <>
-            {fabs?.map(fb => {
-              return <FAB actionButton={fb} size="medium" key={fb.label} />;
-            })}
-          </>
-        </Slide>
-      )}
+      <Collapse
+        style={
+          slot === Slot.PAGE_END
+            ? { textAlign: 'right' }
+            : { textAlign: 'left' }
+        }
```

Comment:
Then we could use it here as well:

```suggestion
        style={{ textAlign: slotOptions[slot].textAlignment }}
```

The benefit is that you see all options then close together and doesn't need more if statements if we add a 3rd option.